### ISSUE TITLE
QT-4 installation fails 

### ISSUE THREAD

**Issue by joeyklee** [OSGeo/homebrew-osgeo4mac#251]
Hi there,

I'm on a 2015 macbook 12in, running osx yosemite 10.10.5
During installation, the process fails at qt-4 installation with the following error:

```
==> Installing osgeo/osgeo4mac/qgis2 dependency: qt-4
==> Using the sandbox
==> Downloading https://download.qt.io/official_releases/qt/4.8/4.8.7/qt-everywh
Already downloaded: /Users/leejoey/Library/Caches/Homebrew/qt-4-4.8.7.tar.gz
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/480b7
Already downloaded: /Users/leejoey/Library/Caches/Homebrew/qt-4--patch-c8a0fa819c8012a7cb70e902abb7133fc05235881ce230235d93719c47650c4e.patch
==> Patching
==> Applying el-capitan.patch
patching file src/gui/painting/qpaintengine_mac.cpp
==> ./configure -prefix /usr/local/Cellar/qt-4/4.8.7 -release -opensource -confi
==> make
Warning: Calling ENV.j1 is deprecated!
Use ENV.deparallelize instead.
/usr/local/Homebrew/Library/Taps/osgeo/homebrew-osgeo4mac/Formula/qt-4.rb:110:in `install'
Please report this to the osgeo/osgeo4mac tap!
```

it seems that calling `ENV.j1` is deprecated. 

Any ideas here?

Thanks!

**Comment by pat-s** [comment]
try a single install of `qt-4` first, the bottle should install without any problems.

`brew install qt-4`

Then `brew install qgis2` should work

**Comment by joeyklee** [comment]
Hi @pat-s 👋 - thanks for the reply. I also tried to install `qt-4` before the qgis installation, but got the same error :/ 

My work machine has some funky admin stuff on it, so it could very well be a result of those artifacts, but that would be a huge shame if that was causing the installation error.

**Comment by pat-s** [comment]
could you post the log of the bottle installation attempt? there should be no `make` step in between which seems to cause the problem when trying to compile from source.

Is the error message really exactly the same when you do `brew install qt-4`?

**Comment by joeyklee** [comment]
Hi @pat-s!
I just gave it another whirl and the output was the same 😩

```
admins-macbook:~ leejoey$ brew tap osgeo/osgeo4mac
==> Tapping osgeo/osgeo4mac
Cloning into '/usr/local/Homebrew/Library/Taps/osgeo/homebrew-osgeo4mac'...
remote: Counting objects: 157, done.
remote: Compressing objects: 100% (141/141), done.
remote: Total 157 (delta 27), reused 38 (delta 1), pack-reused 0
Receiving objects: 100% (157/157), 173.12 KiB | 0 bytes/s, done.
Resolving deltas: 100% (27/27), done.
Checking connectivity... done.
Tapped 68 formulae (170 files, 734.8KB)
admins-macbook:~ leejoey$ brew install qt-4
==> Installing qt-4 from osgeo/osgeo4mac
==> Using the sandbox
==> Downloading https://download.qt.io/official_releases/qt/4.8/4.8.7/qt-everywh
Already downloaded: /Users/leejoey/Library/Caches/Homebrew/qt-4-4.8.7.tar.gz
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/480b7
Already downloaded: /Users/leejoey/Library/Caches/Homebrew/qt-4--patch-c8a0fa819c8012a7cb70e902abb7133fc05235881ce230235d93719c47650c4e.patch
==> Patching
==> Applying el-capitan.patch
patching file src/gui/painting/qpaintengine_mac.cpp
==> ./configure -prefix /usr/local/Cellar/qt-4/4.8.7 -release -opensource -confi
==> make
Warning: Calling ENV.j1 is deprecated!
Use ENV.deparallelize instead.
/usr/local/Homebrew/Library/Taps/osgeo/homebrew-osgeo4mac/Formula/qt-4.rb:110:in `install'
Please report this to the osgeo/osgeo4mac tap!

==> make install
==> /usr/libexec/PlistBuddy -c Add :LSEnvironment:QT_PLUGIN_PATH string "/usr/lo
Error: Operation not permitted - /usr/local/lib/qt-4
admins-macbook:~ leejoey$ 
```

**Comment by joeyklee** [comment]
I should also add it took about 40min to fail

**Comment by pat-s** [comment]
Just read you are on Yosemite, only bottles for Sierra exist.

Hence, I can not reproduce your problem because you are on Yosemite :/ 

I assume it fails because it tries to apply the el-capitan patch and you are running Yosemite -> so this won't work. There might be also other problems around..

Any issues why you do not want to upgrade to Sierra?

**Comment by joeyklee** [comment]
Thanks for your swift reply! It seems you're also in the eurozone.
I haven't updated to sierra, but I will do that and give it a try. I will ping back once ive given it a try. thanks!

**Comment by joeyklee** [comment]
hi! worked like a charm! the update to sierra fixed the problems. Plus the installation time for qt-4 was within seconds or a min, not 40 min. 

thanks for the feedback! I'll close this up if that's cool with you @pat-s ! 🙌

**Comment by joeyklee** [comment]
I may have spoken a bit too soon - now the qgis installation fails, but fails in the same way as #209 

```
c02px172gcn4:~ leejoey$ brew install qgis2
==> Installing qgis2 from osgeo/osgeo4mac
==> Using the sandbox
==> Downloading https://github.com/qgis/QGIS/archive/final-2_18_10.tar.gz
Already downloaded: /Users/leejoey/Library/Caches/Homebrew/qgis2-2.18.10.tar.gz
==> cmake .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/qgis2/2
==> make
==> make install
Last 15 lines from /Users/leejoey/Library/Logs/Homebrew/qgis2/03.make:
-- Installing: /usr/local/Cellar/qgis2/2.18.10/QGIS.app/Contents/MacOS/../Resources/python/qgis/_gui.so
-- Installing: /usr/local/Cellar/qgis2/2.18.10/QGIS.app/Contents/MacOS/../Resources/python/qgis/_server.so
-- Installing: /usr/local/Cellar/qgis2/2.18.10/QGIS.app/Contents/MacOS/../Resources/python/qgis/_analysis.so
-- Installing: /usr/local/Cellar/qgis2/2.18.10/QGIS.app/Contents/MacOS/../Resources/python/qgis/_networkanalysis.so
-- Installing: /usr/local/opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/uic/widget-plugins/qgis_customwidgets.py
CMake Error at python/cmake_install.cmake:119 (file):
  file INSTALL cannot copy file
  "/tmp/qgis2-20170804-24602-1zloy5/QGIS-final-2_18_10/python/custom_widgets/qgis_customwidgets.py"
  to
  "/usr/local/opt/pyqt-qt4/lib/qt-4/python2.7/site-packages/PyQt4/uic/widget-plugins/qgis_customwidgets.py".
Call Stack (most recent call first):
  cmake_install.cmake:41 (include)


make: *** [install] Error 1

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/osgeo/homebrew-osgeo4mac/issues

These open issues may also help:
QGIS2 build fails in pyqt-qt4 make https://github.com/OSGeo/homebrew-osgeo4mac/issues/209
QGIS2  2.18.10 fails to build on OSX Sierra 10.12.5 https://github.com/OSGeo/homebrew-osgeo4mac/issues/243
QGIS2 - could not load "processing" plugin https://github.com/OSGeo/homebrew-osgeo4mac/issues/203
QGIS2 standalone pyqgis app gives SIGSEGV error on quit https://github.com/OSGeo/homebrew-osgeo4mac/issues/232
```

**Comment by pat-s** [comment]
`qgis2` (2.18.10) currently has no bottle. I built one for myself, see #242

`brew install qgis2-ltr` should work for you, it has a bottle and currently runs on v.2.14.14

**Comment by joeyklee** [comment]
Thanks for the tip, though I ended up getting an older version of qgis running, tried the above, and now qgis is no longer working. Any advice on how to start clean on my machine?

**Comment by joeyklee** [comment]
whew! after a bit of macguyvering around and installing dependencies again, I'm up and running with the `qgis2-ltr`. thanks for all the tips @pat-s ! 🙏

**Comment by pat-s** [comment]
glad to hear @joeyklee!

**Comment by dakcarto** [comment]
Compilation on 10.10 should be fixed in 8669bd6
